### PR TITLE
fix kubeadm init when certdir changed

### PR DESCRIPTION
**Pull request by xilabao**
If --cert-dir specified, `kubeadm init`  failed.

### CODE REVIEW

**Comment by dmmcquay** [comment]
@k8s-bot ok to test

**Comment by dmmcquay** [comment]
please write tests for all new functions

**Comment by xilabao** [comment]
@dmmcquay PTAL

**Comment by dmmcquay** [comment]
/lgtm

@luxas can you take a look?

File: cmd/kubeadm/app/master/manifests.go [hunk 1]
```diff
@@ -151,6 +157,22 @@ func WriteStaticPodManifests(cfg *kubeadmapi.MasterConfiguration) error {
 	return nil
 }
 
+func newVolume(cfg *kubeadmapi.MasterConfiguration, name, path string) api.Volume {
```
Comment (luxas) [inline cmd/kubeadm/app/master/manifests.go]: no need to pass cfg here

**Review by luxas — COMMENTED** [review]
LGTM besides one nit, please fix

File: cmd/kubeadm/app/master/manifests.go (hunk 1 above)
Comment (xilabao) [inline cmd/kubeadm/app/master/manifests.go]: all Volume function pass cfg (etc. pkiVolume).  should we remove cfg from them ?

**Comment by xilabao** [comment]
@luxas @dmmcquay PTAL

**Review by dmmcquay — APPROVED** [review]

**Comment by luxas** [comment]
/approve

@k8s-bot cvm gce e2e test this

**Pull request merged**